Plugins/BossBlock.lua

```diff
@@ -187,6 +202,10 @@ do
 		if self.db.profile.blockQuestTrackingTooltips then
 			SetCVar("showQuestTrackingTooltips", questTrackingValue)
 		end
+		if self.db.profile.hideObjectiveTracker and not objectiveTrackerWasCollapsed and IsInRaid() then
```

Comment:
I'd just check `objectiveTrackerWasCollapsed` in this condition to make sure that when we collapsed it, we'll expand it again (in case the other conditions change in between, which is unlikely, but still the cleaner way imo). This also requires setting it to `nil` right after.